dateutil/test/test_parser.py

```diff
@@ -238,6 +238,60 @@ def test_parse_bytearray(self):
         assert res == expected
 
 
+class TestTzinfoInputTypes(object):
+    def test_tzinfo_dict_could_return_none(self):
+        dstr = "2017-02-03 12:40 BRST"
+        result = parse(dstr, tzinfos={"BRST": None})
+        expected = datetime(2017, 2, 3, 12, 40)
+        assert result == expected
+
+    def test_tzinfos_callable_could_return_none(self):
+        dstr = "2017-02-03 12:40 BRST"
+        result = parse(dstr, tzinfos=lambda *args: None)
+        expected = datetime(2017, 2, 3, 12, 40)
+        assert result == expected
+
+    def test_invalid_tzinfo_input(self):
+        dstr = "2014 January 19 09:00 UTC"
+        # Pass an absurd tzinfos object
+        tzinfos = {"UTC": ValueError}
+        with pytest.raises(TypeError):
+            parse(dstr, tzinfos=tzinfos)
+
+    def test_valid_tzinfo_tzinfo_input(self):
+        dstr = "2014 January 19 09:00 UTC"
+        tzinfos = {"UTC": tz.tzutc()}
```

Comment:
I prefer to use `tz.UTC` over `tz.tzutc()` as a general rule in `dateutil` itself. I realize that in `pandas` there are backwards-compatibility problems with that, but in `dateutil`'s test suite that's not the case ;)

Reply:
Changed.

When did tz.UTC become available?  I'll open an issue in pandas to update the min dependencies; tz.UTC will be non-trivially more performant in the cython code.

Reply:
Looks like [2.7.0](https://github.com/dateutil/dateutil/blob/master/NEWS#L266).